geopandas/plotting.py

```diff
@@ -130,21 +276,35 @@ def plot_series(s, cmap='Set1', color=None, ax=None, linewidth=1.0,
     if ax is None:
         fig, ax = plt.subplots(figsize=figsize)
         ax.set_aspect('equal')
-    color_generator = gencolor(len(s), colormap=cmap)
-    for geom in s:
-        if color is None:
-            col = next(color_generator)
-        else:
-            col = color
-        if geom.type == 'Polygon' or geom.type == 'MultiPolygon':
-            if 'facecolor' in color_kwds:
-                plot_multipolygon(ax, geom, linewidth=linewidth, **color_kwds)
-            else:
-                plot_multipolygon(ax, geom, facecolor=col, linewidth=linewidth, **color_kwds)
-        elif geom.type == 'LineString' or geom.type == 'MultiLineString':
-            plot_multilinestring(ax, geom, color=col, linewidth=linewidth, **color_kwds)
-        elif geom.type == 'Point':
-            plot_point(ax, geom, color=col, **color_kwds)
+
+    num_geoms = len(s.index)
+    if color:
+        colors = pd.Series([color] * num_geoms)
+    else:
+        color_generator = gencolor(len(s), colormap=cmap)
+        colors = pd.Series([next(color_generator) for _ in xrange(num_geoms)])
+
+    poly_idx = (s.geometry.type == 'Polygon') | (s.geometry.type == 'MultiPolygon')
+    polys = s.geometry[poly_idx]
+    if not polys.empty:
+        # Plot the fill with default or user-specified alpha, but do not draw outlines.
+        plot_polygon_collection(ax, polys, colors[poly_idx], False, linewidth=0, **color_kwds)
+        # Draw the edges, fully opaque, but no facecolor.
+        edges_kwds = color_kwds.copy()
+        edges_kwds['alpha'] = 1
+        edges_kwds['facecolor'] = 'none'
+        plot_polygon_collection(ax, polys, colors[poly_idx], False, linewidth=linewidth, **edges_kwds)
```

Comment:
The logic is different here depending if it's in `plot_series` or `plot_dataframe`.  (specifically, the `plot_values` argument)